scripts/install

```diff
@@ -9,138 +9,81 @@
 source _common.sh
 source /usr/share/yunohost/helpers
 
-#=================================================
-# MANAGE SCRIPT FAILURE
-#=================================================
-
-# Exit if an error occurs during the execution of the script
-ynh_abort_if_errors
-
-#=================================================
-# RETRIEVE ARGUMENTS FROM THE MANIFEST
-#=================================================
-
-export app=$YNH_APP_INSTANCE_NAME
-# Retrieve arguments
-ynh_export repository passphrase on_calendar conf data apps mailalert
-
 #=================================================
 # STORE SETTINGS FROM MANIFEST
 #=================================================
-server=""
-if [[ $repository == *"@"* ]]; then
-    server=$(echo "$repository" | cut -d"@" -f2 | cut -d"/" -f1)
-    if [[ $server == *":"* ]]; then
-        server="[$(echo "$server" | cut -d":" -f1)]:$(echo "$server" | cut -d":" -f2)"
+
+ssh_regex='^ssh://([^@]*)@([^:/]*)(:[0-9]+)?/(.*)$'
+if [[ "$repository" =~ $ssh_regex ]]; then
+    ssh_user="${BASH_REMATCH[1]}"
+    server="${BASH_REMATCH[2]}"
+    port="${BASH_REMATCH[3]}"
```

Comment:
I am not sure what is the expected value of `port`. Should it be of this form `:1234`? Otherwise, you'd better remove the semi-colon: 
```bash
port="${BASH_REMATCH[3]/:/}"
```

Reply:
Port should contain the colon. You can see why some lines below, and more importantly:
```
server="[$server]$port"
```
We want something like `[mydomain.tld]:22` as explained in `ssh-keygen` doc.